### PR TITLE
Changed - tidied up redis buildout

### PR DESCRIPTION
**Pull request by wengole**
@mattss Just noticed it was sourcing the template from the root and then putting the config there too

### CODE REVIEW

File: redis.cfg [hunk 1]
```diff
@@ -33,12 +33,12 @@ redis-logfile = ${buildout:directory}/var/log/redis.log
...
 cmd = ${redis:location}/src/redis-server
 arguments =
-  ${buildout:directory}/redis.conf
+  ${buildout:bin-directory}/redis.conf
 
 [redis-conf]
 recipe = collective.recipe.template
-input = redis.conf.tpl
-output = redis.conf
+input = buildout-templates/redis.conf.tpl
```
Comment (mattss) [inline redis.cfg]: I suggest:

    ${buildout:directory}/buildout-templates/redis.conf.pl

File: redis.cfg [hunk 2]
```diff
@@ -33,12 +33,12 @@ redis-logfile = ${buildout:directory}/var/log/redis.log
...
+output = ${buildout:bin-directory}/redis.conf
```
Comment (mattss) [inline redis.cfg]: I don't think redis.conf should go in the bin directory. At the root of the project makes more sense.

Comment (mgrbyte) [inline redis.cfg]: fwiw, I usually have these things put at the project root too, 
but then always have to add the generated configuration file(s) to .gitignore too.

Comment (wengole) [inline redis.cfg]: @mattss @mgrbyte I like the idea of having a nice tidy project root with just bootstrap and buildout configs, then mirroring the [Linux Filesystem Hierarchy Standard](http://www.pathname.com/fhs/). Buildout already creates a bin/ and most things create a var/ directory so I'd like to see a etc/ where all the configs go.

I know this goes against the grain (if there is such a thing) of buildouts, but at the very least can we put it in parts/redis/etc/ like supervisor does (for example - parts/supervisor/etc/supervisord.conf) ?

Comment (mgrbyte) [inline redis.cfg]: I'm generally sympathetic to that idea.  
e.g supervisor's recipe generates  parts/<part-name>/supervisord.conf,
I wouldn't be against any template recipe doing the same (e.g parts//<part-name>/<conf-file>. I *think* you can reference this in buildouts with ${<part-name>:location}/<conf-file>.

**Comment by adamcheasley** [comment]
I am all for not putting generated files in the root of the project. That is likely to get messy quite quickly.

**Comment by mattss** [comment]
+1 for putting them in the relevant 'parts' dir

**Comment by wengole** [comment]
@mattss @adamcheasley @mgrbyte Updated PR ...

**Comment by adamcheasley** [comment]
It looks like you've got a 'redis-conf' in parts, which doesn't seem to be defined anyway, then a 'redis' part which is not referenced anywhere.

**Comment by mattss** [comment]
:accept:

**Pull request merged**